pandas/tseries/index.py

```diff
@@ -1629,13 +1632,17 @@ def normalize(self):
         return DatetimeIndex(new_values, freq='infer', name=self.name,
                              tz=self.tz)
 
-    def searchsorted(self, key, side='left'):
-        if isinstance(key, (np.ndarray, Index)):
-            key = np.array(key, dtype=_NS_DTYPE, copy=False)
+    @Substitution(klass='DatetimeIndex')
+    @Appender(_shared_docs['searchsorted'])
+    def searchsorted(self, v, side='left', *args):
+        validate_args(args, min_length=0, max_length=1)
```

Comment:
``pandas`` will not create one, but if you specify it directly using ``np.searchsorted`` then we are passing it thru. At least that's the intention (and looks like code/tests exercise that).  So we are passing thru numpy args that we don't care about, but won't error on them (as its very hard to tell *where* they actually come from), e.g. a user could in theory

``s.searchsorted(....., sorter=[....])``. But since its not in the doc-string, nor signature its not likely.

The exception would be in cases where we are not allowing something (e.g. in the ``np.round`` case, where ``out=`` is passed in intending to be used to do something in-place, which we don't allow via that API). So that *should* raise (if ``out`` is not None, but that's the other issue).